File: knowledge-mobile/modules/gemma-inference/ios/GemmaInferenceModule.swift
import ExpoModulesCore

public class GemmaInferenceModule: Module {
  public func definition() -> ModuleDefinition {
    Name("GemmaInferenceModule")

    Events("onToken")

    AsyncFunction("isAvailable") { (_: String) in
      // TODO: Check MediaPipe model readiness.
      false
    }

    AsyncFunction("checkFeatureStatus") { (_: String) in
      // TODO: Return available | downloading | unavailable
      "unavailable"
    }

    AsyncFunction("generate") { (_: String, prompt: String, _: [String: Any]) in
      // TODO: Wire MediaPipe LlmInference.generateResponseAsync streaming.
      self.sendEvent("onToken", ["token": "[stub-token]"])
      return "Stub iOS response for: \(prompt)"
    }

    AsyncFunction("generateWithImage") { (_: String, prompt: String, _: String) in
      // TODO: base64 -> UIImage -> multimodal inference call.
      return "Stub iOS image response for: \(prompt)"
    }

    AsyncFunction("generateWithAudio") { (_: String, prompt: String, _: String) in
      // TODO: base64 -> PCM audio content.
      return "Stub iOS audio response for: \(prompt)"
    }
  }
}

File: knowledge-mobile/modules/gemma-inference/android/GemmaInferenceModule.kt
package expo.modules.gemmainference

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class GemmaInferenceModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("GemmaInferenceModule")

    Events("onToken")

    AsyncFunction("isAvailable") { model: String ->
      // TODO: Hook ML Kit GenAI Prompt API checkFeatureStatus
      model == "e2b"
    }

    AsyncFunction("checkFeatureStatus") { _: String ->
      // TODO: return available | downloading | unavailable from AICore
      "unavailable"
    }

    AsyncFunction("generate") { _: String, prompt: String, _: Map<String, Any> ->
      // TODO: Use PromptRequest + addOnPartialResponseListener + Dispatchers.IO
      sendEvent("onToken", mapOf("token" to "[stub-token]"))
      "Stub Android response for: $prompt"
    }

    AsyncFunction("generateWithImage") { _: String, prompt: String, _: String ->
      // TODO: Decode base64 to Bitmap and pass as multimodal content
      "Stub Android image response for: $prompt"
    }

    AsyncFunction("generateWithAudio") { _: String, prompt: String, _: String ->
      // TODO: Decode PCM audio and pass as multimodal audio content
      "Stub Android audio response for: $prompt"
    }
  }
}
